securedrop_client/gui/widgets.py

```diff
@@ -692,7 +692,10 @@ def show_sources(self, sources: List[Source]):
             self.empty_conversation_view.show_no_source_selected_message()
             self.empty_conversation_view.show()
 
-        self.source_list.update(sources)
+        deleted_sources = self.source_list.update(sources)
+        for source_uuid in deleted_sources:
```

Comment:
this change makes a lot of sense to me, before we would delete the source items without deleting the conversation wrapper